Marlin/src/HAL/STM32/pinsDebug.h

```diff
@@ -217,58 +204,62 @@ bool pwm_status(const pin_t Ard_num) {
+        #elif defined (PH_0)
+          case 'H' : alt_all = GPIOH->AFR[ind]; break;
+        #elif defined (PI_0)
+          case 'I' : alt_all = GPIOI->AFR[ind]; break;
+        #elif defined (PJ_0)
+          case 'J' : alt_all = GPIOJ->AFR[ind]; break;
+        #elif defined (PK_0)
+          case 'K' : alt_all = GPIOK->AFR[ind]; break;
+        #elif defined (PL_0)
+          case 'L' : alt_all = GPIOL->AFR[ind]; break;
+        #endif
+      }
+      if (over_7) pin_number -= 8;
 
-    uint8_t alt_func = (alt_all >> (4 * pin_number)) & 0x0F;
-    SERIAL_ECHOPAIR("Alt Function: ", alt_func);
-    if (alt_func < 10) SERIAL_CHAR(' ');
-    SERIAL_ECHOPGM(" - ");
-    switch (alt_func) {
-      case  0 : SERIAL_ECHOPGM("system (misc. I/O)"); break;
-      case  1 : SERIAL_ECHOPGM("TIM1/TIM2 (probably PWM)"); break;
-      case  2 : SERIAL_ECHOPGM("TIM3..5 (probably PWM)"); break;
-      case  3 : SERIAL_ECHOPGM("TIM8..11 (probably PWM)"); break;
-      case  4 : SERIAL_ECHOPGM("I2C1..3"); break;
-      case  5 : SERIAL_ECHOPGM("SPI1/SPI2"); break;
-      case  6 : SERIAL_ECHOPGM("SPI3"); break;
-      case  7 : SERIAL_ECHOPGM("USART1..3"); break;
-      case  8 : SERIAL_ECHOPGM("USART4..6"); break;
-      case  9 : SERIAL_ECHOPGM("CAN1/CAN2, TIM12..14  (probably PWM)"); break;
-      case 10 : SERIAL_ECHOPGM("OTG"); break;
-      case 11 : SERIAL_ECHOPGM("ETH"); break;
-      case 12 : SERIAL_ECHOPGM("FSMC, SDIO, OTG"); break;
-      case 13 : SERIAL_ECHOPGM("DCMI"); break;
-      case 14 : SERIAL_ECHOPGM("unused (shouldn't see this)"); break;
-      case 15 : SERIAL_ECHOPGM("EVENTOUT"); break;
+      uint8_t alt_func = (alt_all >> (4 * pin_number)) & 0x0F;
+      SERIAL_ECHOPAIR("Alt Function: ", alt_func);
+      if (alt_func < 10) SERIAL_CHAR(' ');
+      SERIAL_ECHOPGM(" - ");
+      switch (alt_func) {
+        case  0 : SERIAL_ECHOPGM("system (misc. I/O)"); break;
+        case  1 : SERIAL_ECHOPGM("TIM1/TIM2 (probably PWM)"); break;
+        case  2 : SERIAL_ECHOPGM("TIM3..5 (probably PWM)"); break;
+        case  3 : SERIAL_ECHOPGM("TIM8..11 (probably PWM)"); break;
+        case  4 : SERIAL_ECHOPGM("I2C1..3"); break;
+        case  5 : SERIAL_ECHOPGM("SPI1/SPI2"); break;
+        case  6 : SERIAL_ECHOPGM("SPI3"); break;
+        case  7 : SERIAL_ECHOPGM("USART1..3"); break;
+        case  8 : SERIAL_ECHOPGM("USART4..6"); break;
+        case  9 : SERIAL_ECHOPGM("CAN1/CAN2, TIM12..14  (probably PWM)"); break;
+        case 10 : SERIAL_ECHOPGM("OTG"); break;
+        case 11 : SERIAL_ECHOPGM("ETH"); break;
+        case 12 : SERIAL_ECHOPGM("FSMC, SDIO, OTG"); break;
+        case 13 : SERIAL_ECHOPGM("DCMI"); break;
+        case 14 : SERIAL_ECHOPGM("unused (shouldn't see this)"); break;
+        case 15 : SERIAL_ECHOPGM("EVENTOUT"); break;
+      }
     }
-  }
+  #else
+    // TODO: F1 doesn't support changing pins function, so we need to check the function of the PIN and if it's enabled
```

Comment:
Does this work in pinsDebug in the F1 HAL, or was this functionality never present?

Reply:
It's not implemented for maple too, looks like it goes to a generic pinsDebug.

Reply:
There is a `pwm_details`, in HAL/STM32F1, although it doesn't seem to do a lot.